### PR TITLE
CI Opa test at push

### LINKED ISSUE #45: Run `opa test` as part of build
This is easy to setup using the new [setup-opa](https://github.com/open-policy-agent/setup-opa) GitHub Action 😃 

```yaml
name: Run OPA Tests  
on: [push]
jobs:
  Run-OPA-Tests:
    runs-on: ubuntu-latest
    steps:
    - name: Check out repository code
      uses: actions/checkout@v2

    - name: Setup OPA
      uses: open-policy-agent/setup-opa@v1
      with:
        version: latest

    - name: Run OPA Tests 
      run: opa test tests/*.rego -v
```

### PR DESCRIPTION
**Pull request by nissessenap**
Add a OPA test to see if all our OPA rules passes.

Fix: #45

### CODE REVIEW

**Comment by nissessenap** [comment]
@anderseknert could you enlighten me why I should use

```
opa test tests/*.rego -v
```

I read [the github action](https://github.com/open-policy-agent/setup-opa) and it gives the same example you do.

But running opa locally with the latest release I get the following error:

```
opa test docker/opa-policies/test*.rego
11 errors occurred:
docker/opa-policies/test_aws_ecr_repository.rego:43: rego_unsafe_var_error: var authz is unsafe
docker/opa-policies/test_aws_ecr_repository.rego:47: rego_unsafe_var_error: var authz is unsafe
docker/opa-policies/test_aws_ecr_repository.rego:51: rego_unsafe_var_error: var authz is unsafe
docker/opa-policies/test_aws_eks_cluster.rego:43: rego_unsafe_var_error: var authz is unsafe
docker/opa-policies/test_aws_eks_cluster.rego:47: rego_unsafe_var_error: var authz is unsafe
docker/opa-policies/test_aws_eks_cluster.rego:51: rego_unsafe_var_error: var authz is unsafe
docker/opa-policies/test_aws_s3_bucket.rego:43: rego_unsafe_var_error: var authz is unsafe
docker/opa-policies/test_aws_s3_bucket.rego:47: rego_unsafe_var_error: var authz is unsafe
docker/opa-policies/test_aws_s3_bucket.rego:51: rego_unsafe_var_error: var authz is unsafe
docker/opa-policies/test_aws_vpc.rego:43: rego_unsafe_var_error: var authz is unsafe
rego_compile_error: error limit reached
```

But if I instead change over to `opa test docker/opa-policies` everything passes.

I have tested to break one of my test while running `opa test docker/opa-policies` and then I can see a simple fail. So I'm rather sure that it's actually running my tests.

**Comment by simongottschlag** [comment]
> @anderseknert could you enlighten me why I should use
> 
> 
> 
> ```
> 
> opa test tests/*.rego -v
> 
> ```
> 
> 
> 
> I read [the github action](https://github.com/open-policy-agent/setup-opa) and it gives the same example you do.
> 
> 
> 
> But running opa locally with the latest release I get the following error:
> 
> 
> 
> ```
> 
> opa test docker/opa-policies/test*.rego
> 
> 11 errors occurred:
> 
> docker/opa-policies/test_aws_ecr_repository.rego:43: rego_unsafe_var_error: var authz is unsafe
> 
> docker/opa-policies/test_aws_ecr_repository.rego:47: rego_unsafe_var_error: var authz is unsafe
> 
> docker/opa-policies/test_aws_ecr_repository.rego:51: rego_unsafe_var_error: var authz is unsafe
> 
> docker/opa-policies/test_aws_eks_cluster.rego:43: rego_unsafe_var_error: var authz is unsafe
> 
> docker/opa-policies/test_aws_eks_cluster.rego:47: rego_unsafe_var_error: var authz is unsafe
> 
> docker/opa-policies/test_aws_eks_cluster.rego:51: rego_unsafe_var_error: var authz is unsafe
> 
> docker/opa-policies/test_aws_s3_bucket.rego:43: rego_unsafe_var_error: var authz is unsafe
> 
> docker/opa-policies/test_aws_s3_bucket.rego:47: rego_unsafe_var_error: var authz is unsafe
> 
> docker/opa-policies/test_aws_s3_bucket.rego:51: rego_unsafe_var_error: var authz is unsafe
> 
> docker/opa-policies/test_aws_vpc.rego:43: rego_unsafe_var_error: var authz is unsafe
> 
> rego_compile_error: error limit reached
> 
> ```
> 
> 
> 
> But if I instead change over to `opa test docker/opa-policies` everything passes.
> 
> 
> 
> I have tested to break one of my test while running `opa test docker/opa-policies` and then I can see a simple fail. So I'm rather sure that it's actually running my tests.
> 
> 

This is most likely due to the json file not being read when only using *rego

**Review by simongottschlag — APPROVED** [review]
LGTM

File: docker/opa-policies/terraform.rego [hunk 1]
```diff
@@ -92,7 +92,7 @@ resources[resource_type] = all {
     ]
 }
 
-other_resources[resource_type] := all {
```
Comment (anderseknert) [inline docker/opa-policies/terraform.rego:95]: This should work with OPA v0.40.0, and is the preferred operator going forward.

**Review by anderseknert — APPROVED** [review]
@NissesSenap sorry, the path was just an example! Of course you should use whatever makes sense here :)

File: docker/opa-policies/terraform.rego (hunk 1 above)
Comment (simongottschlag) [reply docker/opa-policies/terraform.rego:95]: Ah, we should use `=` instead of `:=` in 0.40.0 and forward? 🤷‍♂️

Comment (anderseknert) [reply docker/opa-policies/terraform.rego:95]: No, the other way around :) `:=` (assignment) is preferred over `=` (unification), for everything but the very few cases where it's actually needed.

Comment (anderseknert) [reply docker/opa-policies/terraform.rego:95]: See https://github.com/open-policy-agent/opa/issues/4555

Comment (simongottschlag) [reply docker/opa-policies/terraform.rego:95]: Thank you @anderseknert!

@NissesSenap maybe we should just update the OPA version instead? Perhaps we should go through the code and make sure we have the same operator everywhere as well.

Comment (anderseknert) [reply docker/opa-policies/terraform.rego:95]: Just to be clear — either will work as expected, so this was very much just a nit :) Mainly just surprised to see it going in this direction. But if compatibility with older versions of OPA is a concern, then that's definitely a valid reason.

Comment (nissessenap) [reply docker/opa-policies/terraform.rego:95]: I will update the other PR so it includes a new version of opa and fix the syntax.

**Pull request merged**